addons/sale_crm/models/crm_lead.py

```diff
@@ -101,3 +101,15 @@ def _merge_get_fields_specific(self):
         # add all the orders from all lead to merge
         fields_info['order_ids'] = lambda fname, leads: [(4, order.id) for order in leads.order_ids]
         return fields_info
+
+    def _update_revenues_from_so(self):
+        for opportunity in self:
+            min_order_untaxed_total = min([
+                order.amount_untaxed
+                for order in opportunity.order_ids.filtered(lambda order: order.state != 'cancel' and order.amount_untaxed
+                    and self.company_id.currency_id == order.currency_id)
+            ], default=0.0)
+
+            if opportunity.expected_revenue < min_order_untaxed_total:
+                opportunity.expected_revenue = min_order_untaxed_total
+                opportunity._message_log(body=_('Expected revenue has been updated based on the linked Sales orders and Quotations.'))
```

Comment:
This should be a tracked field. And it is ok if for some usage we don't really need the tracking, would say better have standard code than manual tracking.